### PR TITLE
feat: add `DatabaseEmitter` implementation class 

### PR DESCRIPTION
**Pull request by AlexPatrie**
**What does this PR do?**:
- `feat`: ports over and implements `DatabaseEmitter` class from `vivarium-core`, complete with all abstract methods in process-bigraph
- `feat`: adds utils functions to `emitter.py` file interacting with aforementioned object.
- `docs`: adds type annotations wherever possible, complete with doc string representation
- `chore`: refactors testing directory and associated files complete with test model files.
- `feat`: adds example processes to `experiments` library

### CODE REVIEW

File: process_bigraph/emitter.py [hunk 1]
```diff
@@ -56,3 +87,539 @@ def query(self, query=None):
             result = self.history
 
         return result
+
+
+class DatabaseEmitter(Emitter):
```
Comment (eagmon) [inline process_bigraph/emitter.py:92]: Emitters are now just Steps, and should inherit from `Step`

**Review by eagmon — COMMENTED** [review]
@AlexPatrie this is fantastic, thank you! Did you get this to work as-is?

One comment -- the `Emitter` base class is a type of `Step` and gets added to the `Composite` just like any other process would, so I think we need to add the `schema` method for this to work. Steps/Processes have inputs that they read and then perform the emit in whatever form. Here is an example of how an emitter gets wired into the composite: https://github.com/vivarium-collective/process-bigraph/blob/14d32a1e7fc34246f91bfe6a500d55f39ad0fcb0/process_bigraph/experiments/minimal_gillespie.py#L212-L223

File: process_bigraph/emitter.py (hunk 1 above)
Comment (eagmon) [reply process_bigraph/emitter.py:92]: Correction: we did make an actual `Emitter` class, which inherits from `Step` and adds a new `query` method: https://github.com/vivarium-collective/process-bigraph/blob/14d32a1e7fc34246f91bfe6a500d55f39ad0fcb0/process_bigraph/emitter.py#L8

Comment (eagmon) [reply process_bigraph/emitter.py:92]: So I think you are good here

File: process_bigraph/emitter.py [hunk 2]
```diff
@@ -56,3 +87,539 @@ def query(self, query=None):
...
+                query: a list of tuples pointing to fields within the experiment data.
+                    In the format: [('path', 'to', 'field1'), ('path', 'to', 'field2')]
+
+            Returns:
+                `Dict`
+        """
+        return get_history_data_db(self.history, self.experiment_id, query)
+
+    def update(self, state):
+        self.emit(state)
```
Comment (eagmon) [inline process_bigraph/emitter.py:198]: nice!

File: process_bigraph/emitter.py (hunk 1 above)
Comment (prismofeverything) [reply process_bigraph/emitter.py:92]: Agreed, let's use `RAMEmitter` or `ConsoleEmitter` as an example, and make a test for the `DatabaseEmitter` that shows it in action.

**Review by prismofeverything — COMMENTED** [review]
Good to get all this in here, thanks @AlexPatrie! Will need to be adapted a bit for the new emitter structure (emitters are just steps now) but it's a good start.

File: process_bigraph/emitter.py (hunk 1 above)
Comment (AlexPatrie) [reply process_bigraph/emitter.py:92]: Got it. I now realize the full contract of abstract methods was not completely signed.

Comment (AlexPatrie) [reply process_bigraph/emitter.py:92]: So I will now implement `schema()` and `query()`!

File: process_bigraph/experiments/copasi_process.py [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+    get_parameters,
+    get_reactions,
+    set_species,
+    run_time_course,
+    get_compartments,
+    model_info
+)
+
+
+class CopasiProcess(Process):
```
Comment (eagmon) [inline process_bigraph/experiments/copasi_process.py:19]: move to core-processes

**Review by eagmon — APPROVED** [review]
This looks good, let's get the core processes out to a separate repo and then this is good to merge. Thanks!

**Pull request merged**